pytket/phir/phirgen_parallel.py

```diff
@@ -0,0 +1,348 @@
+        qubits2groups[qubit] = -4
+    max_len = max(num_scs_per_qubit.values())
+
+    while index < max_len:
+        for qubit in sub_commands:
+            # check to make sure you are not accessing beyond the end of the list
+            if index < num_scs_per_qubit[qubit]:
+                sc = sub_commands[qubit][index]
+                gate = sc.op.type
+                if gate == tk.OpType.Rz:
+                    group_number = rz_group_number
+                    valid_pll_op = True
+                elif gate == tk.OpType.PhasedX:
+                    group_number = r1xy_group_number
+                    valid_pll_op = True
+                else:
+                    group_number = other_group_number
+                    valid_pll_op = False
+                # does a group exist for that gate type?
+                group_available = group_number in groups
+                # is that group later in execution than the
+                # most recent group for an op on that qubit?
+                order_preserved = group_number > qubits2groups[qubit]
+                # is the group size still under the maximum allowed parallel ops?
+                group_size = len(groups[group_number]) if group_number in groups else 0
+                group_not_too_large = group_size < max_parallel_sq_gates
+                # is the op parallelizable (only RZ or R1XY)?
+                if (
+                    group_available
+                    and order_preserved
+                    and group_not_too_large
+                    and valid_pll_op
+                ):
+                    groups[group_number].append(sc)
+                    qubits2groups[qubit] = group_number
+                else:
+                    # make a new group:
+                    if gate == tk.OpType.Rz:
+                        rz_group_number += 3
+                        group_number = rz_group_number
+                    elif gate == tk.OpType.PhasedX:
+                        r1xy_group_number += 3
+                        group_number = r1xy_group_number
+                    else:
+                        other_group_number += 3
+                        group_number = other_group_number
+                    groups[group_number] = [sc]
+                    qubits2groups[qubit] = group_number
+        index += 1
+
+    return dict(sorted(groups.items()))
+
+
+def groups2qops(groups: dict[int, list[tk.Command]], ops: list[dict[str, Any]]) -> None:  # noqa: PLR0912
+    """Convert the groups of parallel ops to properly formatted PHIR."""
+    for group_number in groups:
+        group = groups[group_number]
+        angles2qops: dict[tuple[sympy.Expr | float, ...], dict[str, Any]] = {}
+        comment_insert = None
+        for qop in group:
+            if not qop.op.is_gate():
+                append_cmd(qop, ops)
+            else:
+                comment_insert = f" {tket_gate_to_phir[qop.op.type]}"
```

Comment:
```suggestion
                comment_insert = f"{tket_gate_to_phir[qop.op.type]}"
```
unnecessary space, leading to e.g. `Comment(c='Parallel  R1XY')`

Reply:
that space is necessary, otherwise the PHIR model rejects the string as being too short. the min length is 3 characters.

Reply:
That sounds weirdly arbitrary.

Reply:
I was thinking of comments being helpful, the minimum I could think of was a single qubit gate application such as `X q`. So I will argue that it's a good catch from phir validation. In this PR, we have several kinds of comments:

```py
CVarDefine(metadata=None, data='cvar_define', data_type='u32', variable='c', size=2),
Comment(c=' RZ'),
SQOp(metadata=None, qop='RZ', angles=([1.0], 'pi'), args=[('q', 0), ('q', 1)]),
Comment(c=' R1XY'),
SQOp(metadata=None, qop='R1XY', angles=([0.5, 0.5], 'pi'), args=[('q', 0), ('q', 1)]),
Comment(c=' RZ'),
SQOp(metadata=None, qop='RZ', angles=([1.0], 'pi'), args=[('q', 0), ('q', 1)]),
Comment(c=' R1XY'),
SQOp(metadata=None, qop='R1XY', angles=([1.5, 0.5], 'pi'), args=[('q', 1)]),
Comment(c='ZZPhase(0.5) q[0], q[1];'),
TQOp(metadata=None, qop='RZZ', angles=([0.5], 'pi'), args=[(('q', 0), ('q', 1))]),
MOp(metadata=None, mop='Transport', duration=(0.0, 'ms')),
Comment(c='Parallel  RZ'),
```

I'd think the old behavior, e.g. shown in `ZZPhase(0.5) q[0], q[1];` above, should be retained for non-parallel gates.

Reply:
this was due to the formatting of the comment being different when it was added using the append_cmd function from phirgen.py. I fixed them to be consistent.